internal/controller/etcdopstask/reconcile_task.go

```diff
@@ -0,0 +1,222 @@
+		return ctrlutils.ReconcileWithError(err)
+	}
+	return ctrlutils.ContinueReconcile()
+}
+
+// admitTask checks if the task is in a pending state.
+// If so, it updates the phase to admit and invokes the task handler's Admit method.
+// If the admit operation is in progress, it requeues the task.
+// If the admit operation fails, it updates the task state to Rejected.
+// If the admit operation succeeds, it updates the task.status.state to InProgress.
+// If the task is not in its Pending state, it skips the step.
+func (r *Reconciler) admitTask(ctx context.Context, logger logr.Logger, taskObjKey client.ObjectKey, taskHandler handler.Handler) ctrlutils.ReconcileStepResult {
+	logger = logger.WithValues("step", "admitTask")
+	logger.Info("Executing step: admitTask")
+
+	task, err := r.getTask(ctx, taskObjKey)
+	if err != nil {
+		return ctrlutils.ReconcileWithError(err)
+	}
+	// If the task is not in Pending state, skip the admit step.
+	if task.Status.State != nil && *task.Status.State != druidv1alpha1.TaskStatePending {
+		return ctrlutils.ContinueReconcile()
+	}
+
+	var etcdOpsTaskList druidv1alpha1.EtcdOpsTaskList
+	if err = r.client.List(ctx, &etcdOpsTaskList, client.InNamespace(task.Namespace)); err != nil {
+		return ctrlutils.ReconcileWithError(err)
+	}
+
+	for _, existingTask := range etcdOpsTaskList.Items {
+		if existingTask.Spec.EtcdName != nil && task.Spec.EtcdName != nil &&
+			*existingTask.Spec.EtcdName != "" && *task.Spec.EtcdName != "" &&
+			*existingTask.Spec.EtcdName == *task.Spec.EtcdName {
+			if !existingTask.IsCompleted() && existingTask.Name != task.Name {
+				err := druiderr.WrapError(
+					fmt.Errorf("an EtcdOpsTask for etcd %s/%s is already in progress (task: %s)", existingTask.Namespace, *existingTask.Spec.EtcdName, existingTask.Name),
+					ErrDuplicateTask,
+					string(druidv1alpha1.OperationPhaseAdmit),
+					"duplicate EtcdOpsTask for the same etcd is already in progress",
+				)
+
+				if statusErr := r.updateTaskStatus(ctx, task, taskStatusUpdate{
+					Operation: &druidv1alpha1.LastOperation{
+						Type:        druidv1alpha1.LastOperationTypeReconcile,
+						State:       druidv1alpha1.LastOperationStateError,
+						Description: err.(*druiderr.DruidError).Message,
+					},
+					Phase: ptr.To(druidv1alpha1.OperationPhaseAdmit),
+					State: ptr.To(druidv1alpha1.TaskStateRejected),
+					Error: err,
+				}); statusErr != nil {
+					return ctrlutils.ReconcileWithError(errors.Wrapf(err, "failed to update status: %v", statusErr))
+				}
+				logger.Info("Task rejected due to duplicate task, requeuing after TTL for deletion",
+					"existingTask", existingTask.Name,
+					"ttlSeconds", task.Spec.TTLSecondsAfterFinished,
+					"timeToExpiry", task.GetTimeToExpiry())
+				return ctrlutils.ReconcileAfter(task.GetTimeToExpiry(), "Task rejected, waiting for TTL to expire before deletion")
+			}
+		}
+	}
+
+	if err := r.updateTaskStatus(ctx, task, taskStatusUpdate{
+		Operation: &druidv1alpha1.LastOperation{
```

Comment:
Do you really need to use `LastOperation`?
This is excessive reporting that is leading to way too many calls to the KAPI. Can you please count the number of calls on an average a task is making just to update the status?